### PR TITLE
Upgrade 2 minor versions

### PR DESCRIPTION
**Pull request by areed**

### CODE REVIEW

File: scripts/install.sh [hunk 1]
```diff
@@ -295,7 +295,7 @@ function main() {
     apply_bash_flag_overrides "$@"
     parse_yaml_into_bash_variables
     parse_kubernetes_target_version
-    discover
+    discover full-cluster
```
Comment (emosbaugh) [inline scripts/install.sh:298]: when is it not "full-cluster"?

Comment (areed) [reply scripts/install.sh:298]: in join.sh and upgrade.sh top level scripts

File: scripts/common/discover.sh [hunk 2]
```diff
@@ -116,65 +118,136 @@ detectLsbDist() {
...
+
+KUBERNETES_STEP_REMOTE_PRIMARIES=0
+KUBERNETES_UPGRADE_REMOTE_PRIMARIES_MINOR=0
+KUBERNETES_UPGRADE_REMOTE_PRIMARIES_PATCH=0
+
+KUBERNETES_STEP_SECONDARIES=0
+KUBERNETES_UPGRADE_SECONDARIES_MINOR=0
+KUBERNETES_UPGRADE_SECONDARIES_PATCH=0
+
 discoverCurrentKubernetesVersion() {
```
Comment (emosbaugh) [inline scripts/common/discover.sh:133]: man this function is complex

File: scripts/common/discover.sh [hunk 3]
```diff
@@ -116,65 +118,136 @@ detectLsbDist() {
...
-                KUBERNETES_UPGRADE=1
-                KUBERNETES_UPGRADE_WORKERS_PATCH=1
-            fi
         fi
+    done
+
+    # preflights
+    if [ $MAX_CLUSTER_NODE_MINOR_FOUND -gt $KUBERNETES_TARGET_VERSION_MINOR ]; then
+        printf "%s %s %s" \
+            "the currently installed kubernetes version is 1.${MAX_CLUSTER_NODE_MINOR_FOUND}." \
```
Comment (emosbaugh) [inline scripts/common/discover.sh:235]: i suppose you should capitalize these sentences.

**Review by emosbaugh — APPROVED** [review]

**Pull request merged**